src/rust/lib/graph-editor/src/component/visualization/class.rs

```diff
@@ -154,3 +176,126 @@ impl Visualization {
     }
 }
+
+
+
+// ===========================
+// === Visualization Class ===
+// ===========================
+
+/// Result of the attempt to instantiate a `Visualization` from a `Class`.
+pub type InstantiationResult = Result<Visualization,Box<dyn Error>>;
+
+/// Specifies a trait that allows the instantiation of `Visualizations`.
+///
+/// The `Class` provides a way to implement structs that allow the instantiation of specific
+/// visualizations, while already providing general information that doesn't require an
+/// instantiated visualization, for example, the name or input type of the visualization.
+///
+/// There are two example implementations: The `JsSourceClass`, which is based on a JS snippet to
+/// instantiate `JsRenderer`, and the fairly generic `NativeConstructorClass`, that only requires
+/// a function that can create a InstantiationResult. The later can be used as a thin wrapper around
+/// the constructor methods of native visualizations.
+///
+/// Example
+/// --------
+/// ```no_run
+/// use graph_editor::component::visualization;
+/// use graph_editor::component::visualization::Visualization;
+/// use graph_editor::component::visualization::renderer::example::native::BubbleChart;
+/// use ensogl::display::Scene;
+/// use std::rc::Rc;
+///
+/// // Create a `visualization::Class` from a JS source code snippet.
+/// let js_source_class = visualization::JsSourceClass::from_js_source_raw(r#"
+///
+///    class BubbleVisualization {
+///         static inputTypes = ["[[float;3]]"]
+///         onDataReceived(root, data) {}
+///         setSize(root, size) {}
+///     }
+///
+///     return BubbleVisualization;
+///
+/// "#.into());
+///
+/// // Create a `visualization::Class` that instantiates a `BubbleChart`.
+/// let native_bubble_vis_class = visualization::NativeConstructorClass::new(
+///     visualization::ClassAttributes {
+///         name        : "Bubble Visualization (native)".to_string(),
+///         input_types : vec!["[[float;3]]".to_string().into()],
+///     },
+///     |scene:&Scene| Ok(Visualization::new(BubbleChart::new(scene)))
+/// );
+/// ```
+pub trait Class: Debug {
+    /// Provides additional information about the `Class`, for example, which `DataType`s can be
+    /// rendered by the instantiated visualization.
+    fn attributes(&self) -> &ClassAttributes;
+    /// Create new visualization, that is initialised for the given scene. This can fail if the
+    /// `visualization::Class` contains invalid data, for example, JS code that fails to execute,
+    /// or if the scene is in an invalid state.
+    // TODO consider not allowing failing here and require the checking on instantiation of the `Class`.
+    // TODO consider not providing the scene here, but hooking the the shapes/dom elements into the
+    // scene externally.
+    fn instantiate(&self, scene:&Scene) -> InstantiationResult;
+}
+
+#[derive(Clone,Debug,Default)]
+#[allow(missing_docs)]
+pub struct ClassHandle {
```

Comment:
I know we are using allow missing docs for the inner field, but docs regarding what actually this Handle is for should be there :)